### DISCUSSION TITLE
Stacking QSE

### DISCUSSION THREAD

**Discussion by hametar0u** [unitaryfund/mitiq#2267] [Q&A] [answered]
Just following up on our conversation in #2251 , can nothing can be stacked on top of QSE? If I wanted to use the `mitigate_executor` function with it, then the circuit has to return a `MeasurementResult` type object which I don't think is the same as the density matrix. And if I were to put QSE at the back, then ZNE or something would somehow have to return a density matrix, and I don't think that's possible?

**Comment by natestemen** [comment, marked as answer]
If you want to stack 2 QEM techniques which require different signatures for the required executor, you can sometimes still stack them if you can convert from one type to another.

Suppose you have an executor with signature `ex: circuit -> density matrix`. When you apply Quantum Subspace Expansion with `mitiq.qse.execute_with_qse`, it will return a float. You can think of `execute_with_qse(ex, -)`[^1] as an executor with signature `circuit -> float` which can then be fed into ZNE.

[^1]: Where `-` represents the "argument" to this function.
